### PR TITLE
HTML Spinner Hotfix

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Adds a new data attribute to the HTML Spinner API that allows submit buttons to skip the `disabled` behavior.

image

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in HTML
 - [x] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)

image

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)
  
### Accessibility Testing
  - [x] Axe browser extension
  - [ ] Jaws Inspect
  - [x] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [ ] Design Team
 - [ ] Accessibility Team

### CODE REVIEW

**Review by guern1kn — APPROVED** [review]

File: html/components/spinners.js [hunk 1]
```diff
@@ -84,6 +90,37 @@ const spinners = () => {
       }
     });
   });
+
+  /**
+   * This selector is exactly the same as data-sprk-spinner="click" except
+   * the disabled attribute will not be added to the button when it is
+   * spinning.
+   */
+  getElements('[data-sprk-spinner="is-not-disabled"]', (spinnerContainer) => {
```
Comment (RVMendoza) [inline html/components/spinners.js:99]: is this implying that it either has `click` or `is-not-disabled`? 

What are you thoughts on if we grabbed elements with [data-sprk-spinner] and then do an in statement based on if its value is `click` or `is-not-disabled`so we're repeating less code?

Comment (RVMendoza) [reply html/components/spinners.js:99]: Perhaps something like this: https://stackoverflow.com/questions/7084557/select-all-elements-with-a-data-xxx-attribute-without-using-jquery

and then using `getAttribute` to switch between the two

File: html/components/spinners.js [hunk 2]
```diff
@@ -32,14 +32,20 @@ const setSpinning = (element, options) => {
...
   el.classList.add('sprk-c-Button--has-spinner');
   el.setAttribute('aria-label', spinningAriaLabel);
   el.setAttribute('data-sprk-spinner-text', el.textContent);
-  el.setAttribute('disabled', '');
   el.setAttribute('data-sprk-has-spinner', 'true');
   el.setAttribute('style', `width: ${width}px`);
 
+  // This flag should be used for submit buttons so that
+  // the disabled attribute does not suppress the submit behavior.
+  if (!doNotDisable) {
```
Comment (RVMendoza) [inline html/components/spinners.js:45]: this double negative is throwing me off, perhaps something more brain friendly like `hasDoNotDisable`

Comment (RVMendoza) [reply html/components/spinners.js:45]: it's only *slightly* less confusing though so if you have a better idea, go for it haha

**Review by RVMendoza — APPROVED** [review]
these changes are great! I'm glad we could find a more efficient way of doing the same thing!

**Pull request merged**